### ISSUE TITLE
false positive for sc2189 when using bash '|&' redirection

### ISSUE THREAD

**Issue by wbolster** [koalaman/shellcheck#1051]
#### For bugs
- Rule Id (if any, e.g. SC1000): sc2189
- My shellcheck version (`shellcheck --version` or "online"):  0.4.6
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [ ] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/usr/bin/env bash

example() {
    echo stdout
    echo stderr > /dev/stderr
}

example |& > /dev/null
```

#### Here's what shellcheck currently says:

```
$ shellcheck myscript
 
Line 8:
example |& > /dev/null
           ^-- SC2189: You can't have | between this redirection and the command it should apply to.
```

#### Here's what I wanted or expected to see:

Nothing. This is from the bash man page:

```
If |& is used, command's standard error, in addition to its standard
output, is connected to command2's standard input through the pipe; it
is shorthand for 2>&1 |. This implicit redirection of the standard
error to the standard output is performed after any redirections
specified by the command.
```

**Comment by jleffler** [comment]
The `|&` is a variation on `|`.  It needs to appear where `|` would appear, between commands.  `command1 |& command2` is a shorthand for `command1 2>&1 | command2`.  Your example use is bogus; the `>/dev/null` isn't associated with any command.  If you had `cat` or `:` or almost anything associated with that redirection, you'd be in with a chance: `example |& > /dev/null :` (I/O redirection before the command is legal, if confusing).

**Comment by wbolster** [comment]
ah, thanks for explaining and enlightening me. :)  will close now.
